tools/GenerateDocFxStructure/Program.cs

```diff
@@ -0,0 +1,313 @@
+        /// Process given file in give folder. Check all references.
+        /// </summary>
+        /// <param name="folder">Folder where file live.</param>
+        /// <param name="filepath">Complete path of the file to check.</param>
+        private static void ProcessFile(DirectoryInfo folder, string filepath)
+        {
+            if ((_options == null) ||(_message == null))
+            {
+                throw new ArgumentNullException($"Options or message can't be null, something went wrong");
+            }
+
+            string[] lines = File.ReadAllLines(filepath);
+            // Create the destination file
+            var relativeDestinationFilePath = filepath.Substring(_options.SourceFolder!.Length + 1);
+            var newFileDestination = Path.Combine(_options.DestinationFolder!, relativeDestinationFilePath);
+            // Ensure the path exists and is created
+            var dir = Path.GetDirectoryName(newFileDestination);
+            if (!Directory.Exists(dir))
+            {
+                Directory.CreateDirectory(dir!);
+            }
+
+            StreamWriter destinationFile = File.CreateText(newFileDestination);
+
+            // get matches per line to be able to reference a line in a file.
+            int linenr = 1;
+            foreach (string line in lines)
+            {
+                string lineToWrite = line;
+                Regex rxLine = new Regex(@"(\[[^\]]+\]{1}\({1}[^\)]+\){1})");
+                MatchCollection matches = rxLine.Matches(line);
+                if (matches.Any())
+                {
+                    // process all matches
+                    foreach (Match match in matches)
+                    {
+                        // get just the reference
+                        int start = match.Value.IndexOf("](") + 2;
+                        string relative = match.Value.Substring(start);
+                        int end = relative.IndexOf(")");
+                        relative = relative.Substring(0, end);
+                        string initialReletive = relative;
+
+                        // relative string contain not only URL, but also "title", get rid of it
+                        int positionOfLinkTitle = relative.IndexOf('\"');
+                        if (positionOfLinkTitle > 0)
+                        {
+                            relative = relative.Substring(0, relative.IndexOf('\"')).Trim();
+                        }
+
+                        // strip in-doc references using a #
+                        if (relative.Contains("#"))
+                        {
+                            relative = relative.Substring(0, relative.IndexOf("#"));
+                        }
+
+                        // decode possible HTML encoding
+                        relative = HttpUtility.UrlDecode(relative);
+
+                        // check link if not to a URL, in-doc link or e-mail address
+                        if (!relative.StartsWith("http:") &&
```

Comment:
should you warn on unrecognized link type?

Reply:
There is a warning later on. So if malformed here, you'll get an error later on

Reply:
example!
![image](https://user-images.githubusercontent.com/8145578/127871212-e5b4addc-fca3-4acc-95aa-d5e8f65133e8.png)